Sources/App/Controllers/ClientController.swift

```diff
@@ -0,0 +1,52 @@
+struct ClientController {
+  
+  func addRoutes(to droplet: Droplet) {
+    droplet.get("user", User.parameter, "client", handler: index)
```

Comment:
We can use associated type for fix this case. For example: create BaseChildController, which should have parent as BaseController.